src/perl/lib/wtsi_clarity/dao/study_dao.pm

```diff
@@ -76,21 +74,35 @@ has 'manager' => (
 );
 sub _build_manager {
   my $self = shift;
-  my $users = ();
-
-  foreach my $study_user_id (@{$self->study_user_ids}) {
-    my $study_user_dao = wtsi_clarity::dao::study_user_dao->new( lims_id => $study_user_id);
-    if ($study_user_dao) {
-      my $study_user_data = ();
-      $study_user_data->{'login'} = $study_user_dao->login;
-      $study_user_data->{'email'} = $study_user_dao->email;
-      $study_user_data->{'name'} = $study_user_dao->name;
-
-      push @{$users}, $study_user_data;
-    }
+  my @users = map { $self->get_message($_) } @{$self->study_user_ids};
+  return to_json(\@users);
+}
+
+has 'cost_code' => (
+  isa => 'Str',
+  is => 'ro',
+  traits => [ 'DoNotSerialize' ],
+  lazy_build => 1,
+);
+sub _build_cost_code {
+  my $self = shift;
+  return $self->findvalue($COST_CODE_PATH);
+}
+
+sub get_message {
```

Comment:
Could you rename it to get_user_message?
I think, it is more meaningful, because it is in the study_dao and not in the study_user_dao...